core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveDeltaFloatingCodec.java

```diff
@@ -282,6 +288,12 @@ public void decodeAndFillVector(byte[] pageData, ColumnVectorInfo vectorInfo, Bi
           for (int i = 0; i < size; i += DataTypes.INT.getSizeInBytes()) {
             vector.putFloat(rowId++, (max - ByteUtil.toIntLittleEndian(pageData, i)) / floatFactor);
           }
+        } else if (pageDataType == DataTypes.LONG) {
```

Comment:
Why we are handling long here for float data type, for float type page data type cannot be long.

Reply:
For AdaptiveFloating codec, float data was coming as long for complex primitive.
So added for delta flow also.

Reply:
this means , In some scenario it's storing float data as Long [4bytes to 8 bytes]? Can u pls check the test once

Reply:
I have tested, With current UT itself it is hitting. for [Null, 5.512] it is using long as storage for complex primitive adaptive. Base behavior needs to check. I guess it can be analyzed separately

Reply:
For this, I have checked 
If No complex type, (if it is just primitive type) same values goes to DirectCompress, not adaptive. But for complex primitive it goes to adaptive because of below code. And as min max is stored as double precision. Long is chosen for this.


`DefaultEncodingFactory#selectCodecByAlgorithmForFloating()`

```
} else if (decimalCount < 0 && !isComplexPrimitive) {
      return new DirectCompressCodec(DataTypes.DOUBLE);
    } else {
      return getColumnPageCodec(stats, isComplexPrimitive, columnSpec, srcDataType, maxValue,
          minValue, decimalCount, absMaxValue);
    }
```

I don't know (remember) why complex primitive should not enter direct compress.  why that check is explicitly added.

Reply:
Can u pls raise a jira for above issue

Reply:
jira is created 

https://issues.apache.org/jira/browse/CARBONDATA-3965